src/ray/core_worker/core_worker.h

```diff
@@ -100,10 +100,14 @@ class CoreWorker {
 
   void SetActorId(const ActorID &actor_id) {
     RAY_CHECK(actor_id_.IsNil());
+    absl::MutexLock lock(&mutex_);
     actor_id_ = actor_id;
   }
 
-  void SetWebuiDisplay(const std::string &message) { webui_display_ = message; }
+  void SetWebuiDisplay(const std::string &message) {
+    absl::MutexLock lock(&mutex_);
+    webui_display_ = message;
```

Comment:
@pcmoritz I think this was a thread safety issue.

Reply:
Yeah, I think you are right!

Reply:
We should move this out of the header file and add lock annotations?